samza-core/src/main/java/org/apache/samza/operators/spec/OperatorSpecs.java

```diff
@@ -53,7 +53,7 @@ private OperatorSpecs() {}
    * @return  the {@link StreamOperatorSpec}
    */
   public static <M, OM> StreamOperatorSpec<M, OM> createMapOperatorSpec(
-      MapFunction<M, OM> mapFn, MessageStreamImpl<OM> nextStream, int opId) {
+      MapFunction<? super M, ? extends OM> mapFn, MessageStreamImpl<OM> nextStream, int opId) {
```

Comment:
Will be a little cleaner to do these casts in MessageStreamImpl and leave these as K, V, M